core/types/tx_bls_test.go

```diff
@@ -34,24 +33,33 @@ func createEmptyBLSTxInner(nonce uint64, sk *bls.SecretKey) *BLSTx {
 		Gas:       25000,
 		To:        common.Address{0x03, 0x04, 0x05},
 		Value:     uint256.NewInt(99),
-		Data:      msg,
+		Data:      make([]byte, 50),
 		PublicKey: bls.PublicFromSecretKey(sk),
-		Signature: sig,
 	}
 }
 
 // Test to see if BLS signer works.
 func TestBLSTxSigning(t *testing.T) {
+	// Mimic wallet level
 	k, err := crypto.GenerateBLSKey()
 	if err != nil {
 		t.Fatal("error creating keys:", err)
 	}
+
+	// Construct tx
 	tx, err := createEmptyBLSTx(k)
 	if err != nil {
 		t.Fatal("error creating empty BLSTx:", err)
 	}
 	hash := tx.Hash()
+
+	// Mimic wallet signing
+	sig := bls.SignatureToBytes(bls.Sign(k, hash.Bytes()))
+	tx.SetSignature(sig)
```

Comment:
From the EIP: The signature value signature is calculated by constructing a BLS signature over the following digest:

`tx_hash = keccak256(BLS_TX_TYPE || rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, to, value, data, access_list, sender]))`

Furthermore, this `signature` field will get unset after being included in a block